<?xml version="1.0"?>
<robot name="red_robot" xmlns:xacro="http://www.ros.org/wiki/xacro"> 
  <!-- Constants for robot dimensions -->
  <xacro:property name="PI" value="3.1415926535897931"/>

  <!-- Import Rviz colors -->
  <xacro:include filename="$(find red_robot_base)/urdf/materials.xacro" />

  <!-- Base Link -->
  <link name="base_link">
    <visual name="top_plate">
      <origin xyz="-0.02 0 0.05" rpy="0 0 0"/>
      <geometry>
	    <box size=".16 .11 .01"/>
      </geometry>
      <material name="red"/>
    </visual>
    <visual name="caster">
      <origin xyz="-0.09 0.0 0.01" rpy="0 0 0"/>
      <geometry>
        <sphere  radius="0.01"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>

  <link name="left_wheel">
    <collision name="collision">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
         <cylinder  length="0.03" radius="0.03"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
        <cylinder  length="0.03" radius="0.03"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  
  <joint type="revolute" name="left_wheel_joint">
    <origin xyz="0 0.05 0.03" rpy="0 0 0"/>
    <child link="left_wheel">left_wheel</child>
    <parent link="base_link">base_link</parent>
    <axis xyz="0 1 0"/>
    <limit effort="100" velocity="100.0"  lower="-5000" upper="5000" />
  </joint> 

  <link name="right_wheel">
    <collision name="collision">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
         <cylinder  length="0.03" radius="0.03"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
        <cylinder  length="0.03" radius="0.03"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  
  <joint type="revolute" name="right_wheel_joint">
    <origin xyz="0 -0.05 0.03" rpy="0 0 0"/>
    <child link="right_wheel">right_wheel</child>
    <parent link="base_link">base_link</parent>
    <axis xyz="0 1 0"/>
    <limit effort="100" velocity="100.0"  lower="-5000" upper="5000" />
  </joint>


  <link name="laser">
    <visual>
      <origin xyz="0 0.0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.03"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <joint type="fixed" name="laser_to_base_joint">
    <origin rpy="0 0 0" xyz="0 0 0.05"/>
    <child link="laser"/>
    <parent link="base_link"/>
  </joint>

</robot>
